source/develop/release-management/features/storage/multipath-events.html.md

```diff
@@ -0,0 +1,174 @@
+Once the host is disconnected from the DC, Vdsm will unregister from
+the 'udev' events.
+
+
+### New vdsm API
+
+New multipath event type
+
+
+#### Multipath Event
+
+An event with information about the multipath device, path and status.
+
+A mapping with these keys:
+- type: Path Failed/Path Reinstated (DM_ACTION)
+- device (GUID): unique GUID of this device (DM_NAME)
+- valid_paths - number of valid paths (DM_NR_VALID_PATHS)
+- seqnum - event sequence number, may be used to detect missed messages later (DM_SEQNUM)
+
+
+### Vdsm multipath event monitoring
+
+Introduce a new Vdsm internal service for handling events:
+
+- Code interested in receiving mutlipath events will subscribe a callback to receive the event
+- When an event is received, all subscribers will receive the events (in another thread).
+- If nobody is registered for events, the events are dropped
+- The multipath event monitor will be started when connecting to the pool
+- The multipath event monitor will be stopped when disconnecting from the pool
+
+
+A Udev event monitor is introduced, listening to device-mapper udev events.
+The relevant events for this feature are PATH_FAILED and PATH_REINSTATED:
+
+Variable Name: DM_ACTION
+Uevent Action(s): KOBJ_CHANGE
+Type: string
+Description:
+Value: Device-mapper specific action that caused the uevent action.
+	PATH_FAILED - A path has failed.
+	PATH_REINSTATED - A path has been reinstated.
+
+- The Vdsm will suscribe the monitor to events during startup.
+- The Vdsm will start the Udev event monitor when connecting to the storage pool.
+- The listener will send an event to the engine when its callback is called.
+- The Vdsm will unregister the listener to the Udev event monitor when 
+disconnecting to the storage pool.
+- The user can disable this feature in a specific host in Vdsm configuration file
+
+In the future, the monitor can be used for other code, for example maintaining LVM filter.
+
+### New engine API
+
+New audit logs are added:
+- Multipath device path up
+- Multipath device path down
+
+## Benefit to oVirt
+
+- The user will be able to detect storage issues faster.
+
+
+## User Experience
+
+New audit logs are added with information on the device path status.
+
+
+## Installation/Upgrade
+
+This feature does not affect engine or Vdsm installation or upgrades
+
+
+## User work-flows
+
+- Once a user connects to an ISCSI server, 'Multipath device path up' event
+will be created.
+
+- If there is an issue with the connection to the storage server, and the 
+multipath reports an issue with one or more device path,
+'Multipath device path down' will be created.
+
+- Once the issue is fixed, 'Multipath device path up' event will be created.
+
+
+## Dependencies / Related Features
+
+Additional package dependency in Vdsm:
+- pyudev
+
+No related features.
+
+
+### Entity Description
+
+NA
+
+## CRUD
+
+NA
+
+
+## Event Reporting
+
+- 'Multipath device path up'
+- 'Multipath device path down'
+
+
```

Comment:
Don't we have an issue increasing the database size a lot by keeping millions of new events?
Do we have policy for dropping old audit logs from the database?